homeassistant/helpers/script.py

```diff
@@ -220,287 +236,349 @@ def referenced_entities(self):
         self._referenced_entities = referenced
         return referenced
 
-    def run(self, variables=None, context=None):
+    def run(self, variables=None, context=None, logger=None, message_base=None):
         """Run script."""
         asyncio.run_coroutine_threadsafe(
-            self.async_run(variables, context), self.hass.loop
+            self.async_run(variables, context, logger, message_base), self.hass.loop
         ).result()
 
-    async def async_run(
-        self, variables: Optional[Sequence] = None, context: Optional[Context] = None
-    ) -> None:
-        """Run script.
+    class _ScriptRun:
+        def __init__(
+            self,
+            hass: HomeAssistant,
+            parent: "Script",
```

Comment:
Let's name it what it is: `script`.

Reply:
That's not a name, it's a type hint. And since Python doesn't support forward references I had to use a string instead of the class name itself.

Reply:
I meant the parameter/instance name. 
```suggestion
            script: "Script",
```

Reply:
Ah, ok.